nms/app/packages/magmalte/app/state/PolicyState.js

```diff
@@ -62,3 +67,48 @@ export async function SetPolicyState(props: Props) {
     setPolicies(newPolicies);
   }
 }
+
+type ProfileProps = {
+  networkId: network_id,
+  qosProfiles: {[string]: policy_qos_profile},
+  setQosProfiles: ({[string]: policy_qos_profile}) => void,
+  key: string,
+  value?: policy_qos_profile,
+};
+
+export async function SetQosProfileState(props: ProfileProps) {
```

Comment:
While I like how convenient this function is in abstracting away the API calls, I would recommend adding a docstring here. It wasn't obvious to me that setting `value` to `null` would trigger a deletion, for example.

Reply:
Not passing in the value would trigger a deletion. Yes. It can be a little unintuitive. I will add comments here.